wlib/RepositoryView.h

```diff
@@ -371,8 +371,17 @@ class CRepositoryViewT
     }
     void DoMoveAttribute(IAttributeVector & attrs, const std::_tstring & target)
     {
-        DoMoveAttributeNoRefreshTarget(attrs, target);
-        TreeNode::RefreshChildren(target, m_Root);
+        if (DoMoveAttributeNoRefreshTarget(attrs, target)) {
+            TreeNode::RefreshChildren(target, m_Root);
+            std::map<std::wstring, bool> modLabels;
+            for (IAttributeVector::const_iterator itr = attrs.begin(); itr != attrs.end(); ++itr)
+            {
+                std::_tstring from = itr->get()->GetModuleQualifiedLabel();
+                if (modLabels.find(from) == modLabels.end())
+                    TreeNode::RefreshChildren(from, m_Root);
+                modLabels[from] = true;
+            }
+        }
```

Comment:
You will need to refresh on failure now as well (as some may have moved successfully?)

Reply:
@GordonSmith yes, you are right! Will add...

Reply:
@dehilsterlexis did this get added?

Reply:
@GordonSmith yes, this was added in my latest push.